SConstruct

```diff
@@ -1247,6 +1247,23 @@ def inkscapeArgs( imageOptions, svg ) :
 	width = int( round( svgObjectInfo["width"] ) )
 	height = int( round( svgObjectInfo["height"] ) )
 
+	# Ensure images are properly pixel aligned and optionally, a specific size.
+	# Transparent container objects should be used where the artwork is of a shape that precludes this.
+
+	if imageOptions.get( "validatePixelAlignment", True ):
+		if width != svgObjectInfo["width"] or height != svgObjectInfo["height"] :
+			raise RuntimeError(
+				"Object with id '%s' is not aligned to pixel boundaries w: %s h: %s" %
+					( id_, svgObjectInfo["width"], svgObjectInfo["height"] )
+			)
```

Comment:
Is it possible to have something where width and height are integers, but we're still not aligned to the pixel grid because the origin is non-integer?

Reply:
I don't think so, as it anchors `0,0` at the start of the object, not the document grid.